src/resolvers/projectResolver.ts

```diff
@@ -256,63 +320,95 @@ export class ProjectResolver {
+      // .leftJoinAndSelect('project.donations', 'donations')
+      .leftJoinAndSelect('project.users', 'users')
+      .leftJoinAndMapOne(
+        'project.adminUser',
+        User,
+        'user',
+        'user.id = CAST(project.admin AS INTEGER)',
+      )
+      .innerJoinAndSelect('project.categories', 'c')
+      .where(
+        `project.statusId = ${ProjStatus.active} AND project.listed = true`,
+      );
+
+    // Filters
+    query = ProjectResolver.addCategoryQuery(query, category);
+    query = ProjectResolver.addSearchQuery(query, searchTerm);
+    query = ProjectResolver.addFilterQuery(
+      query,
       filterBy?.field,
       filterBy?.value,
     );
+    query = ProjectResolver.addUserReaction(query, connectedWalletUserId, user);
+
+    if (orderBy.field === 'traceCampaignId') {
+      // TODO: PRISMA will fix this, temporary fix inverting nulls.
+      const traceableDirection = {
+        ASC: 'NULLS FIRST',
+        DESC: 'NULLS LAST',
+      };
+      query.orderBy(
+        `project.${orderBy.field}`,
+        orderBy.direction,
+        // @ts-ignore
+        traceableDirection[orderBy.direction],
+      );
+    } else {
+      query.orderBy(`project.${orderBy.field}`, orderBy.direction);
+    }
+
+    const [projects, totalCount] = await query
+      .take(take)
+      .skip(skip)
+      .getManyAndCount();
 
     return { projects, totalCount, categories };
   }
 
   @Query(returns => TopProjects)
   async topProjects(
-    @Args() { take, skip, orderBy, category }: GetProjectsArgs,
+    @Args()
+    { take, skip, orderBy, category, connectedWalletUserId }: GetProjectsArgs,
+    @Ctx() { req: { user } }: MyContext,
   ): Promise<TopProjects> {
     const { field, direction } = orderBy;
     const order = {};
     order[field] = direction;
 
-    let projects;
-    let totalCount;
-
-    if (!category) {
-      [projects, totalCount] = await this.projectRepository.findAndCount({
-        take,
-        skip,
-        order,
-        relations: ['reactions'],
-        where: {
-          status: {
-            id: ProjStatus.active,
-          },
-        },
-      });
-    } else {
-      [projects, totalCount] = await this.projectRepository
-        .createQueryBuilder('project')
-        .innerJoin(
-          'project.categories',
-          'category',
-          'category.name = :category',
-          { category },
-        )
-        .innerJoin('project.reactions', 'reaction')
-        .where('project.statusId = 5 AND project.listed = true')
-        .orderBy(`project.${field}`, direction)
-        .limit(skip)
-        .take(take)
-        .innerJoinAndSelect('project.categories', 'c')
-        .getManyAndCount();
+    let query = this.projectRepository.createQueryBuilder('project');
+    // .innerJoin('project.reactions', 'reaction')
+    if (category) {
+      query = query.innerJoin(
```

Comment:
To avoid duplicated code we can reuse the query chaining you moved and declared on the projectResolver.ts.

addCategoryQuery

Reply:
I didn't get the point!

Reply:
Done, thanks for meticulous review